src/rate.rs

```diff
@@ -802,6 +802,32 @@ impl RCState {
     }
     dropped
   }
+
+  pub fn needs_trial_encode(&self, fti: usize) -> bool {
+      self.target_bitrate > 0 && self.nframes[fti] == 0
```

Comment:
An obvious future improvement is to zero this for all frame types when we insert a detected scene change,  but that can be separate.

Reply:
Presumably this would work the same for force keyframes (supposedly forthcoming!).

Reply:
A keyframe that's not at a natural scene change might not need a recode, because its stats are similar to the last keyframe. Dunno if that's worth taking advantage of though.

Reply:
I'm not actually sure that would be strictly true. There's also the case where a keyframe may be forced on what would have been selected to be a "scene change" anyway.